### PR TITLE
[SPARK-47089][BUILD][TESTS] Migrate `mockito 4` to `mockito 5`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to migrate `mockito 4` to `mockito 5`.

### Why are the changes needed?
This is the prerequisite behavior for upgrading `scalatest mockito` to version `3.2.18`.
`Scalatest mockito 3.2.18` no longer supports `mockito 4`.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @dongjoon-hyun @LuciferYang

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocatorSuite.scala [hunk 1]
```diff
@@ -124,10 +124,12 @@ class ExecutorPodsAllocatorSuite extends SparkFunSuite with BeforeAndAfter {
     when(podsWithNamespace.withName(driverPodName)).thenReturn(driverPodOperations)
     when(podsWithNamespace.resource(any())).thenReturn(podResource)
     when(podsWithNamespace.withLabel(anyString(), anyString())).thenReturn(labeledPods)
-    when(podsWithNamespace.withLabelIn(anyString(), any())).thenReturn(labeledPods)
```
Comment (panbingkun) [inline resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocatorSuite.scala:127]: image
image

**Review by dongjoon-hyun — APPROVED** [review]
Oh, thank you so much, @panbingkun . It looks good to me.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 4.0.0.

**Comment by LuciferYang** [comment]
late LGTM ~ thanks @panbingkun and @dongjoon-hyun

**Pull request closed without merging**